File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/publishAndSubscribe/signal/SignalOld.java
package com.pubnub.docs.publishAndSubscribe.signal;

import com.pubnub.api.PubNubException;
import com.pubnub.api.java.PubNub;
import com.pubnub.api.models.consumer.PNPublishResult;
import com.pubnub.docs.SnippetBase;

public class SignalOld extends SnippetBase {

    private void signalMessageToChannel() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#signal-a-message-to-a-channel-1

        PubNub pubNub = createPubNub();

        // snippet.signalMessageToChannel
        pubNub.signal()
                .message("Hello everyone!")
                .channel("bar")
                .async(result -> {
                    result.onSuccess((PNPublishResult signalResult) -> {
                        Long timetoken = signalResult.getTimetoken(); // signal message timetoken
                    }).onFailure((PubNubException exception) -> {
                        System.out.println("error happened while calling signal: " + exception.getMessage());
                    });
                });
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/publishAndSubscribe/signal/SignalOld.kt
package com.pubnub.docs.publishAndSubscribe.signal

import com.pubnub.docs.SnippetBase

class SignalOld : SnippetBase() {
    private fun signalMessageToChannel() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#basic-usage-12

        val pubnub = createPubNub()

        // snippet.signalMessageToChannel
        pubnub.signal(
            message = "hello",
            channel = "my_channel"
        ).async { result ->
            result.onFailure { exception ->
                // Handle error
            }.onSuccess { value ->
                // Handle successful method result
            }
        }
        // snippet.end
    }
}
